ultralytics/nn/tasks.py

```diff
@@ -454,7 +460,7 @@ def loss(self, batch, preds=None):
         return sum(loss.values()), torch.as_tensor([loss[k].detach() for k in ['loss_giou', 'loss_class', 'loss_bbox']],
                                                    device=img.device)
 
-    def predict(self, x, profile=False, visualize=False, batch=None, augment=False):
+    def predict(self, x, profile=False, visualize=False, batch=None, augment=False, embed=None):
```

Comment:
would it be better to use a empty list as default value? as the following code i.e `m.i in embed` is assuming it's a list.

Reply:
even we do pass a list in the top level api, but users might encounter error by calling this `predict` themselves

Reply:
@Laughing-q yes I wanted to use `embed=[]` in the method arguments, but then PyCharm gave a warning here:
<img width="999" alt="Screenshot 2023-12-22 at 02 36 18" src="https://github.com/ultralytics/ultralytics/assets/26833433/76bccee2-f584-4343-bc58-4b4bedb3dc84">

Reply:
ok I see, then None is ok I think, just found there's a `if embed` in front of `m[i] in embed`😂, so there'd be no error even user directly call this function without pass a embed arg.

Reply:
another question, why we need to create another variable `embeddings` to keep embedding rather than reuse `y`? as `y` is not used by default.

Reply:
Yeah that's interesting, maybe I can reuse `y`, I'll take a look.

Reply:
@Laughing-q ok I've used y now in https://github.com/ultralytics/ultralytics/pull/7098/commits/1ff083ed124240511d882af765e8a7e8e1606638

Reply:
@Laughing-q nevermind, this caused issues because `y` values need to be used later in the model for shortcuts etc, so I couldn't save those indices as embeddings. I could have modified slightly but would have added another for-loop.

Reply:
@glenn-jocher yes I know but we can directly use the kept embedding in `y` to return instead of creating a new `embeddings` to append it again.
see https://github.com/ultralytics/ultralytics/pull/6971/files#diff-e2650ac65b6525f6d9c0594a226066d396afbcba637c054d29abe04e62f410c4